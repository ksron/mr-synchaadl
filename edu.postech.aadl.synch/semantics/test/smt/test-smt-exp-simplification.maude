set show timing off .

load ../../interpreter-symbolic.maude

select BOOL-EXP-SIMPLIFICATION .
red b(0) and ([false] or b(0)) .
red [true] and b(2) or (b(4) === [false]) .
red b(0) === b(0) ? [true] and b(2) : b(3) .
red [true] .
red not [true] .
red [true] and [false] .
red [true] xor [false] .
red [true] or [false] .
red [true] implies [false] .
red [true] === [false] .
red [true] =/== [false] .

select REAL-EXP-SIMPLIFICATION .
red ([3] * [3/4]) < r(0) .
red r(1) > r(1) ? [2] / [5] : [1] * [3] + [6] .
red [1] + [2] + [3] + (- [2]).
red [1] + [1] .
red [1] * [1] .
red [1] - [1] .
red [1] / [1] .

red [1] < [1] .
red [1] <= [1] .
red [1] > [1] .
red [1] >= [1] .
