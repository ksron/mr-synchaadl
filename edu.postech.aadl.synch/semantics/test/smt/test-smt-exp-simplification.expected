==========================================
reduce in BOOL-EXP-SIMPLIFICATION : b(0) and (b(0) or [false]) .
rewrites: 2
result SMTBoolVar: b(0)
==========================================
reduce in BOOL-EXP-SIMPLIFICATION : b(2) and [true] or b(4) === [false] .
rewrites: 2
result BoolExp: not b(4) or b(2)
==========================================
reduce in BOOL-EXP-SIMPLIFICATION : b(0) === b(0) ? b(2) and [true] : b(3) .
rewrites: 2
result BoolExp: [true] ? b(2) : b(3)
==========================================
reduce in BOOL-EXP-SIMPLIFICATION : [true] .
rewrites: 0
result SMTBoolValue: [true]
==========================================
reduce in BOOL-EXP-SIMPLIFICATION : not [true] .
rewrites: 2
result SMTBoolValue: [false]
==========================================
reduce in BOOL-EXP-SIMPLIFICATION : [true] and [false] .
rewrites: 2
result SMTBoolValue: [false]
==========================================
reduce in BOOL-EXP-SIMPLIFICATION : [true] xor [false] .
rewrites: 2
result SMTBoolValue: [true]
==========================================
reduce in BOOL-EXP-SIMPLIFICATION : [true] or [false] .
rewrites: 2
result SMTBoolValue: [true]
==========================================
reduce in BOOL-EXP-SIMPLIFICATION : [true] implies [false] .
rewrites: 4
result SMTBoolValue: [false]
==========================================
reduce in BOOL-EXP-SIMPLIFICATION : [true] === [false] .
rewrites: 2
result SMTBoolValue: [false]
==========================================
reduce in BOOL-EXP-SIMPLIFICATION : [true] =/== [false] .
rewrites: 2
result SMTBoolValue: [true]
==========================================
reduce in REAL-EXP-SIMPLIFICATION : [3] * [3/4] < r(0) .
rewrites: 3
result BoolExp: [9/4] < r(0)
==========================================
reduce in REAL-EXP-SIMPLIFICATION : r(1) > r(1) ? [2] / [5] : [6] + [1] * [3] .
rewrites: 7
result RealExp: [false] ? [2/5] : [9]
==========================================
reduce in REAL-EXP-SIMPLIFICATION : [1] + [2] + - [2] + [3] .
rewrites: 7
result SMTRealValue: [4]
==========================================
reduce in REAL-EXP-SIMPLIFICATION : [1] + [1] .
rewrites: 2
result SMTRealValue: [2]
==========================================
reduce in REAL-EXP-SIMPLIFICATION : [1] * [1] .
rewrites: 2
result SMTRealValue: [1]
==========================================
reduce in REAL-EXP-SIMPLIFICATION : [1] - [1] .
rewrites: 2
result SMTRealValue: [0]
==========================================
reduce in REAL-EXP-SIMPLIFICATION : [1] / [1] .
rewrites: 2
result SMTRealValue: [1]
==========================================
reduce in REAL-EXP-SIMPLIFICATION : [1] < [1] .
rewrites: 2
result SMTBoolValue: [false]
==========================================
reduce in REAL-EXP-SIMPLIFICATION : [1] <= [1] .
rewrites: 2
result SMTBoolValue: [true]
==========================================
reduce in REAL-EXP-SIMPLIFICATION : [1] > [1] .
rewrites: 2
result SMTBoolValue: [false]
==========================================
reduce in REAL-EXP-SIMPLIFICATION : [1] >= [1] .
rewrites: 2
result SMTBoolValue: [true]
Bye.
